plugin-core/src/main/java/appland/actions/PluginStatus.java

```diff
@@ -0,0 +1,114 @@
+        if (projectDir == null) return;
+
+        new Task.Backgroundable(project, "Generating AppMap Plugin status report...",  true, PerformInBackgroundOption.ALWAYS_BACKGROUND) {
+
+            private volatile VirtualFile statusReportFile;
+
+            @Override
+            public void run(@NotNull ProgressIndicator indicator) {
+                String reportText = statusReportText(indicator);
+
+                statusReportFile = WriteAction.computeAndWait(() -> {
+                    try {
+                        var file = projectDir.createChildData(this, STATUS_REPORT_FILENAME);
+                        VfsUtil.saveText(file, reportText);
+                        return file;
+                    } catch (IOException e) {
+                        LOG.error(e);
+                        return null;
+                    }
+                });
+            }
+
+            @Override
+            public void onSuccess() {
+                if (statusReportFile != null) {
+                    var editorManager = FileEditorManager.getInstance(project);
+                    editorManager.openFile(statusReportFile, true);
+                }
+            }
+        }.queue();
+    }
+
+    @NotNull
+    private static String statusReportText(ProgressIndicator indicator) {
+        String header = "# AppMap Plugin Status Report\n\n";
+        String cliReport = appmapBinaryStatusReport(CliTool.AppMap);
+        String scannerReport = appmapBinaryStatusReport(CliTool.Scanner);
+        return header + JavaAgentStatus.generateStatusReport(indicator) + cliReport + scannerReport;
+    }
+
+    @NotNull
+    private static String appmapBinaryStatusReport(CliTool type) {
+        var service = AppLandDownloadService.getInstance();
+
+        String latestVersion;
+        try {
+            latestVersion = service.fetchLatestRemoteVersion(type);
+        } catch (IOException e) {
+            latestVersion = null;
+        }
+
+        String latestVersionText = String.format("Latest version: %s",
+                latestVersion == null ? "Failed to check for the latest version" : latestVersion);
+
+        var downloadedVersion = service.findLatestDownloadedVersion(type);
+        var downloadPath = service.getDownloadFilePath(type);
+
+        String downloadedVersionText = "Your version: ";
+        downloadedVersionText += downloadedVersion == null ? "" : downloadedVersion;
+        downloadedVersionText += downloadPath == null ? "" : String.format("\n\nDownload location: %s", downloadPath);
+
+        return getBinaryReportHeader(type) + downloadedVersionText + "\n\n\n" + latestVersionText + "\n\n";
+    }
+
+    @Nullable
+    private static String getBinaryReportHeader(CliTool type) {
+        if (type == CliTool.AppMap) {
+            return "### AppMap CLI Status\n\n";
+        } else if (type == CliTool.Scanner) {
+            return "### AppMap Scanner Status\n\n";
+        }
+
+        return null;
```

Comment:
Personally, I prefer `switch`:
```suggestion
switch (type) {
            case AppMap:
                return "### AppMap CLI Status\n\n";
            case Scanner:
                return "### AppMap Scanner Status\n\n";
            default:
                return null;
        }
```

Reply:
Yeah, that's nicer, I changed it to this